2.6.2. release notes (#1019)
portworx/pxdocs: content/reference/release-notes/portworx.md

series: release-notes
---

## 2.6.2

<!-- December 2, 2020 -->

### Improvements

Portworx has upgraded or enhanced functionality in the following areas:

| **Improvement Number** | **Improvement Description** |
|----|----|
| PWX-17154 | We are adding support for NO_PROXY environment variable with licensing code, that defines which hosts will _not_ use http-proxy for licensing actions.
In addition, `PX_HTTP_PROXY` and `PX_HTTPS_PROXY` environment variables will be _ignored_ when using license-servers and Floating licenses, unless one specifies `PX_FORCE_HTTP_PROXY=1` environment variable (i.e. we will assume local-access when working with Floating licenses). | 
| PWX-16410 | Adding support for K3s v1.19 | 

> | PWX-16738 | Introducing a new feature to enable or disable periodic cache flush on all nodes in the cluster.| 

Reviewer: Introducing new cluster option to enable and disable cache flush on all nodes in cluster. Put that option here @vjs-px

Reviewer 2: Do we really need to advertise this?

Reviewer: IF we don't want to take it out.

Author: removed it

> | PWX-15234 | New feature. If credential object was deleted in k8 secrets when there are pending delete requests , px would try to delete them in the periodic cleanup routine as these delete requests are saved in kvdb. This allows users to delete these pending references to credentials from kvdb.| 

Reviewer: Take out New Feature.

Reviewer 3: No. This was to solve customer scenario. Pls dont take this out. Here is the User Impact and resolution
User impact:  Users may see  periodic alerts about failure to delete cloudsnaps with error messages indicating  credential/secrets not being found.
Resolution: Added new pxctl command that allows users to delete these pending cloudsnap-delete requests from KVDB.  /opt/pwx/bin/pxctl cred delete-refs <name or uuid>

| PWX-16602 | Added an rt_opt to disable zero detection for converting zero filled buffer to discard. |

> | PWX-14705 | It was not possible to pass a "sender" for the email alerts.

Reviewer: Added capability to set the "Sender" for the email alerts as follows.

We now added this ability by passing --sender like.
sv email set --sender. In the same vein we also couldn't pass multiple recipients. We now added the ability to pass multiple emails using a semi-colon like: `sv email set --recipient="email1@portworx.com;email2@portworx.com"`	|
| PWX-16678 | Added PX alerts when PX sets the max storage nodes per zone setting that is different from what the user provided |

> | PWX-16205 | Added a new command to transfer for a Portworx cloud driveset from one storage node to another storageless node. Currently supported only for Google Cloud Platform. Not supported when Portworx is installed using an internal KVDB. |

Reviewer: Put doc link for this

Reviewer: @nathan-px ^^

| PWX-16655 | the `pxctl sv pool cache status <pool>` command is now supported in all operational modes. The following CLI pxctl commands have been moved to only be supported in "pool maintenance" mode and deprecated from "maintenance" mode<ul><li>`pxctl sv pool cache flush <pool id>`</li><li>`pxctl sv pool cache enable <pool id>`</li><li>`pxctl sv pool cache disable <pool id>`</li><li>`pxctl sv pool cache configure <pool id>`</li></ul> |

> | PWX-15693 | - Added support for draining/removing volume attachments from a node through the `pxctl service node drain-attachments` command. | 

Reviewer: Add doc link for this

Reviewer: @nathan-px ^^^

### Fixes

The following issues have been fixed:

|**Issue Number**|**Issue Description**|
|----|----|

> | PWX-16137 | A race-condition between Portworx and Docker occurred at startup. <br/><br/>**User impact:** If the node cannot install NFS-packages (i.e. air-gapped host environments), and runs Portworx with the Docker container-runtime, users might encounter a race-condition on node reboot that leads to Docker creating new local volumes instead of using existing portworx-volumes. <br/><br/>**Resolution:**  |

Reviewer: Resolution : Portworx now detects and rejects docker creating new local volume on node reboot.

Reviewer: @nathan-px ^^^

| PWX-16136 | Portworx pods hung when the `dbus` service became unresponsive.<br/><br/>**User impact:** In situations where the `dbus` service running on the host became unresponsive, the Portworx pod wasn't able to upgrade Portworx or propagate configuration changes. <br/><br/>**Resolution:** We now detect when the `dbus` service is unresponsive and fail over to different methods to run the Portworx upgrades or change configuration. |

> | PWX-16413 | Fixes portworx installation warning on the Ubuntu 20.04 nodes | 

Reviewer: @zoxpx Please format this release not to be more descriptive like above.

Reviewer 4: _"PWX-16413 :: Removes harmless warning during the installation of Portworx.
User impact: during the installation of Portworx into the staging area, the installation would raise installation warning when setting up the services.  These warnings were safe to ignore, but caused unnecessary confusion.
Resolution: fixed the installation process so the warnings would not be raised."_

| PWX-16420 | Portworx pods running on Ubuntu/20.04 nodes did not proxy the the `portworx-output.service` logs to the OCI-Monitor display. <br/><br/>**User impact:** On Ubuntu/20.04, the Portworx pods did not proxy the Portworx service logs. This made troubleshooting difficult on K8s deployments that do not provide SSH access to the hosts. <br/><br/>**Resolution:** The Portworx pods now correctly proxy the Portworx service logs. |
| PWX-16418 | When running on Kubernetes nodes using the ContainerD container runtime, Portworx installs and upgrades sometimes failed.<br/><br/>**User impact:** If a Portworx installation or upgrade was interrupted on the nodes using the ContainerD container runtime, further attempts to install or upgrade Portworx sometimes failed until the node was rebooted.<br/><br/>**Resolution:** Portworx now performs a more thorough cleanup before each install and upgrade operation, solving this issue. |

> | PWX-16853 | Multipart license expiration was not clear. <br/><br/>**User impact:** When running Portworx on a multipart license (i.e. a license composed out of several parts/ActivationIDs), different parts of the license would expire at different times. The `pxctl status` and `pxctl license list` didn't provide any indication that some parts of the license would expire sooner. <br/><br/> **Resolution:** The `pxctl` command now displays a notice if a part of the license will expire sooner than the overall license. |

Reviewer 4: In _"different parts of the license would expire at different times"_  please correct ~would~ to _could_

> | PWX-14214 | Verified with 2.6.2 and operator 1.4.1 <!-- @zox  --> | 

Reviewer: @zoxpx Proper RL note ??

Reviewer 4: it's a minor cleanup issue on node-wipe -- IMO not worthy of Release-node

Reviewer: @nathan-px take it out from RL

| PWX-16769 | License updates sometimes failed to propagate across all nodes. <br/><br/>**User impact:** For users with ETCD as the KVDB and aggressive ETCD compactions, some of the nodes in Portworx cluster skipped the automatic application of updated licenses. Users had to restart the Portworx service to pick up the changes.<br/><br/>**Resolution:** Portworx no longer skips license updates when aggressive ETCD compactions are used.| 

> | PWX-16793 | This fix speeds up Portworx installations for hosts that did not have NFS services preinstalled. <br/><br/>**User impact:** Some host environments may have a large amount of package repositories or have a slow internet connection, which could lead into timeouts while installing NFS services during Portworx installation or upgrade, which ultimately results in inability to use SharedV4 volumes that depend on NFS. <br/><br/>**Resolution:** Portworx installs have been sped up and avoid timeouts with NFS services installations. |

Reviewer: Speed up

Author: "sped up" is the correct term here.

Reviewer 4: Suggest: _"Portworx installs have been optimized to minimize timeouts with NFS service installations"_

| PWX-17057 | With Linux kernel 4.20.x and above, NFS pool stats are incremented at much slower pace than an hour or it may not increment at all if there are no exports. <br/><br/>**User impact:** With Portworx versions prior to 2.6.2, there may have been false alarms about all NFS threads being busy when there were no exports. <br/><br/>**Resolution:** Portworx no longer processes the NFS pool stats when there are no NFS exports. |

> | PWX-16775 | Reduce ListObject timings with google object-store. This resulted in cloudsnaps commands timing out when the bucket involved in cloudsnaps had lot of objects. |

Reviewer: @talakad Need proper text like other RL notes for this.

Reviewer 3: PWX-16775: User impact:  Cloudsnap commands with google object store may timeout when the bucket contains either lot or large cloudsnaps.
Resolution: Fixed Enumerate code that was unnecessarily spending time in iterating over all objects.

Author: I think we can remove this, 16775 was listed in 2.6.1.5

> | PWX-16796 | Proxy username and password were ignored as part of `PX_HTTP_PROXY` on Portworx essentials causing license renewal to fail. <br/><br/>**User impact:** Portworx essentials clusters went into "license expired" mode with PX_HTTP_PROXY <br/><br/>**Resolution:** Portworx essentials now honors Username and Password fields given as part of `PX_HTTP_PROXY` to successfully make connections with proxy |

Reviewer 4: May want to capitalize "Portworx Essentials"   (it's a special license SKU / PX release, not a generic/regular "essentials")

Reviewer: @nathan-px ^^^

> | PWX-16072 | When Portworx was installed in PAYG mode, the cluster license  expired after being unable to reach to billing server instead of going into maintenance mode. | 

Reviewer: Portworx PAYG node used to get into license expiry instead of maintenance mode during unreachabiiity
User Impact: PAYG node license gets expired and need to recommission the node
Resolution : Portworx PAYG node now enter maintenance mode correctly when its unreachable.

Reviewer 4: ~being unable~ -> "not being able" or "failing to" ?
Also might want to check the document and get rid of "  "  (double spaces)

| PWX-16429 | Adding a new drive using `pxctl service drive add` was failing due to an issue with applying labels on the new pool. <br/><br/>**User impact:** If users wanted to add a new Portworx pool to the node, their command to add a new drive using `pxctl service drive add` failed with an error message about labels being too long. This prevented users from creating new pools. <br/><br/>**Resolution:** The Kubernetes labels are being skipped to be applied to the backend storage pools. |

> | PWX-16432 | Multipathd configuration files were not correctly set for blacklisting px devices <br/><br/>**User impact:** Incorrect entries in the Multipathd configuration file caused other devices to be not handled correctly on the host. <br/><br/>**Resolution:** Portworx no longer handles the Multipathd conf file by default and allows you to add blacklist entries to the Multipathd conf file correctly through the runc install argument `-enable-mpcfg-update` |

Reviewer: Take out this as its already released noted in lower 2.6.1.x releases

| PWX-16206 | Portworx failed to correctly detect the max storage nodes per zone when running on GKE. <br/><br/>**User impact:** When running on GKE with autoscaling enabled, Portworx did not detect the preferred value to use for max storage nodes per zone for it's cloud drives. As a result, Portworx would run without or with an incorrect max storage nodes per zone value. This resulted in issues when the cluster size was scaled and unintended nodes became become storage nodes. <br/><br/>**Resolution:** The calculation for max storage nodes per zone based on the GKE autoscaling min pool size values has been fixed. In addition to this, if the minimum number of nodes in the cluster is lower than total number of zones, at least one Portworx node will now be made storage node in a given zone. |
| PWX-16554 | An incorrect check prevented HA level restoration for volumes with HA level 3 under some conditions during decommission operation.<br/<br/> **User impact:** Decommission operation failed to restore HA level for affected volumes if the volume had HA level 3 under some conditions. For such volumes, restore HA level using the pxctl CLI command for ha-update. | 

> | PWX-16407 | Fixed issue with stalled CloudMigration when there was a crash during backup creation | 

Reviewer: @disrani-px Can you please give proper RL note with the following format 
Issue summary
User Impact
Resolution

Reviewer: Issue Summary : DR migration stays in **"Active"** state forever
User impact : Further DRs were waiting forever without finishing
Resolution : Fix Migration indicating incorrect status in certain corner cases where px restarted during migration start

| PWX-16495 | Currently, the KVDB lock is held over the Inspect API. If the remote node didn't respond, Portworx held the KVDB lock for more than 3 minutes and the node where the Attach was issued asserted. <br/><br/> **User impact:** The Portworx service could assert and restart if it tried to remotely detach a volume from a peer node but the request to do that took more than 3 minutes. <br/><br/>**Resolution:** Portworx will not assert and restart if such a request gets stuck or does not return within a given timeout. |
| PWX-13527 | Internal kvdb startup would fail with the error : "Operation cannot be fulfilled on configmaps". This would usually require a wipe of the node and re-install. <br/><br/>**Resolution**: Portworx will now detect such errors received from Kubnernetes and will retry the operation instead of exiting.	| 

> | PWX-15770 | Do not retry NFS mount in a loop on failure. The timeout for NFS unmount command is set to 2 minutes, if retried in a loop an API Mount request could end up taking more than 4 minutes. Verification: This was found in a torpedo job and have not been since since the fix, considering fixed. Will re-open this ticket if accured again. |

Reviewer: No need to release note as this is internal bug

| PWX-16465 | <br/><br/>**User impact:** Encrypted volume attach operations sometimes failed and caused Portworx to assert. The attach call risked getting stuck in the "dm-crypt" layer or in an HTTPS API to the KMS provider.<br/><br/>**Resolution:** Portworx will no longer assert if any calls get stuck in the "dm-crypt" layer or if the HTTPS calls to the KMS providers timeout. |
| PWX-15043  | For NFS Proxy Volumes in the `pxctl volume inspect` command output display the "Mount Options" field. <br/><br/>**User Impact:** the `pxctl volume inspect` command output didn't show the "Mount Options" field for NFS proxy volumes, even if the mount options were explicitly provided while creating such volume. | 
| PWX-16386 | On certain slower systems, a sharedv4 volume wasn't mounted over NFS as soon as it was exported on the server. <br/><br/>**User impact:** Portworx showed the `access denied by server` error. <br/><br/>**Resolution** Portworx now detects this error scenario and retries the NFS mount. | 

> | PWX-17217 | Portworx failed to exit maintenance mode after `drive add` operations showed as "done". <br/><br/>**User impact:** If a user restarted, the `add drive status` command returned “done” while the md reshape was still in progress. Even when the reshape was done, the in-core status of the mountpoint didn't change, and users couldn't exit maintenance mode.<br/><br/>**Resolution:** If you're stuck in maintenance mode as a result of this issue, you can restart Portworx to clear it. | 

Reviewer 2: This is not really a fix. It was marked as "errata" - to be fixed in future.

Reviewer: @nathan-px anyway to say errata

series: release-notes
---

## 2.6.2

December 7, 2020

### New features

> * [Announcing a new command](/reference/cli/cloud-drives-asg/#transfer-cloud-drives-to-a-storageless-node) for transferring a Portworx cloud driveset from one storage node to a storageless node. This command is currently supported only for Google Cloud Platform, and is not supported when Portworx is installed using an internal KVDB. 

Author: @vilasd Not sure if we want to frame these as "new features" or just improvements.

This one and node drain seem like they could be presented as new features if we want to, depending on how much we want to play them up.

### Improvements

Portworx has upgraded or enhanced functionality in the following areas:

| **Improvement Number** | **Improvement Description** |
|----|----|

> | PWX-17154 | Portworx now supports a `NO_PROXY` environment variable with licensing code that defines which hosts will _not_ use http-proxy for licensing actions. In addition, `PX_HTTP_PROXY` and `PX_HTTPS_PROXY` environment variables will be _ignored_ when using license-servers and Floating licenses, unless you specify the `PX_FORCE_HTTP_PROXY=1` environment variable (i.e. Portworx will assume local-access when working with Floating licenses). | 

Reviewer 5: ```suggestion
| PWX-17154 | Portworx now supports a `NO_PROXY` environment variable with a licensing code that defines which hosts will _not_ use an HTTP proxy for licensing actions. In addition, `PX_HTTP_PROXY` and `PX_HTTPS_PROXY` environment variables will be _ignored_ when using license-servers and floating licenses, unless you specify the `PX_FORCE_HTTP_PROXY=1` environment variable (i.e. Portworx will assume local-access when working with Floating licenses). | 
```

| PWX-16410 | Portworx now supports K3s v1.19. | 
| PWX-15234 | You can now delete pending references to credentials from the KVDB.| 

> | PWX-16602 | Portworx now allows for an rt_opt to disable zero detection for converting zero filled buffer to discard. |

Reviewer 5: ```suggestion
| PWX-16602 | Portworx now allows for a runtime option to disable zero detection for converting zero-filled buffer to discard. |
```

| PWX-14705 | You can now set the "Sender" for the email alerts as follows: `sv email set --recipient="email1@portworx.com;email2@portworx.com"`.	|

> | PWX-16678 | Portwox now displays alerts when it sets a different "max storage nodes per zone" setting from what the user provided. |

Reviewer 5: ```suggestion
| PWX-16678 | Portwox now displays alerts when it overrides the user-provided value for the `maxStorageNodesPerZone` parameter. |
```

> | PWX-16655 | Portworx now supports the `pxctl sv pool cache status <pool>` command in all operational modes. The following CLI pxctl commands have been moved to only be supported in "pool maintenance" mode and deprecated from "maintenance" mode:<ul><li>`pxctl sv pool cache flush <pool-id>`</li><li>`pxctl sv pool cache enable <pool-id>`</li><li>`pxctl sv pool cache disable <pool-id>`</li><li>`pxctl sv pool cache configure <pool-id>`</li></ul> |

Reviewer 5: ```suggestion
| PWX-16655 | Portworx now supports the `pxctl service pool cache status <pool>` command in all operational modes. The following CLI `pxctl` commands have been moved to only be supported in "pool maintenance" mode and deprecated from "maintenance" mode:<ul><li>`pxctl service pool cache flush <pool-id>`</li><li>`pxctl service pool cache enable <pool-id>`</li><li>`pxctl service pool cache disable <pool-id>`</li><li>`pxctl service pool cache configure <pool-id>`</li></ul> |
```

> | PWX-15693 | Portworx now allows you to [drain/remove volume attachments](/reference/cli/service/#drain-volume-attachments) from a node through the `pxctl service node drain-attachments` command. | 

Reviewer 6: This can be under a new feature/command

### Fixes

The following issues have been fixed:

|**Issue Number**|**Issue Description**|
|----|----|

> | PWX-16137 | A race-condition between Portworx and Docker occurred at startup. <br/><br/>**User impact:** If the node cannot install NFS-packages (i.e. air-gapped host environments), and runs Portworx with the Docker container-runtime, users might encounter a race-condition on node reboot that leads to Docker creating new local volumes instead of using existing portworx-volumes. <br/><br/>**Resolution:** Portworx now detects and prevents Docker from creating new local volumes on node reboot.|

Reviewer 5: ```suggestion
| PWX-16137 | A race-condition between Portworx and Docker occurred at startup. <br/><br/>**User impact:** If the node cannot install NFS-packages (i.e. air-gapped host environments), and runs Portworx with the Docker container-runtime, users might encounter a race-condition on node reboot that leads to Docker creating new local volumes instead of using existing Portworx volumes. <br/><br/>**Resolution:** Portworx now detects and prevents Docker from creating new local volumes on node reboot.|
```

> | PWX-16136 | Portworx pods hung when the `dbus` service became unresponsive.<br/><br/>**User impact:** In situations where the `dbus` service running on the host became unresponsive, the Portworx pod wasn't able to upgrade Portworx or propagate configuration changes. <br/><br/>**Resolution:** We now detect when the `dbus` service is unresponsive and fail over to different methods to run the Portworx upgrades or change configuration. |

Reviewer 5: ```suggestion
| PWX-16136 | Portworx pods hung when the `dbus` service became unresponsive.<br/><br/>**User impact:** In situations where the `dbus` service running on the host became unresponsive, the Portworx pod wasn't able to upgrade Portworx or propagate configuration changes. <br/><br/>**Resolution:** Portworx now detects when the `dbus` service is unresponsive and fails over to different methods to run the Portworx upgrades or change configuration. |
```

| PWX-16413 | A harmless warning was sometimes displayed during Portworx installation. <br/><br/>**User impact:** When installed into the staging area, Portworx displayed a warning when setting up the services; these warnings were safe to ignore, but caused unnecessary confusion. <br/><br/>**Resolution:** These warnings are no longer displayed. | 

> | PWX-16420 | Portworx pods running on Ubuntu/20.04 nodes did not proxy the the `portworx-output.service` logs to the OCI-Monitor display. <br/><br/>**User impact:** On Ubuntu/20.04, the Portworx pods did not proxy the Portworx service logs. This made troubleshooting difficult on K8s deployments that do not provide SSH access to the hosts. <br/><br/>**Resolution:** The Portworx pods now correctly proxy the Portworx service logs. |

Reviewer 5: ```suggestion
| PWX-16420 | Portworx pods running on Ubuntu/20.04 nodes did not proxy the `portworx-output.service` logs to the OCI-Monitor display. <br/><br/>**User impact:** On Ubuntu/20.04, the Portworx pods did not proxy the Portworx service logs. This made troubleshooting difficult on K8s deployments that do not provide SSH access to the hosts. <br/><br/>**Resolution:** The Portworx pods now correctly proxy the Portworx service logs. |
```

| PWX-16418 | When running on Kubernetes nodes using the ContainerD container runtime, Portworx installs and upgrades sometimes failed.<br/><br/>**User impact:** If a Portworx installation or upgrade was interrupted on the nodes using the ContainerD container runtime, further attempts to install or upgrade Portworx sometimes failed until the node was rebooted.<br/><br/>**Resolution:** Portworx now performs a more thorough cleanup before each install and upgrade operation, solving this issue. |

> | PWX-16853 | Multipart license expiration was not clear. <br/><br/>**User impact:** When running Portworx on a multipart license (i.e. a license composed out of several parts/ActivationIDs), different parts of the license could expire at different times. The `pxctl status` and `pxctl license list` didn't provide any indication that some parts of the license would expire sooner. <br/><br/> **Resolution:** The `pxctl` command now displays a notice if a part of the license will expire sooner than the overall license. |

Reviewer 5: ```suggestion
| PWX-16853 | Multipart license expiration was not clear. <br/><br/>**User impact:** When running Portworx on a multipart license (i.e. a license composed out of several parts/ActivationIDs), different parts of the license could expire at different times. The `pxctl status` and `pxctl license list` commands didn't provide any indication that some parts of the license would expire sooner. <br/><br/> **Resolution:** The `pxctl status` and `pxctl license list` commands now display a notice if a part of the license will expire sooner than the overall license. |
```

| PWX-16769 | License updates sometimes failed to propagate across all nodes. <br/><br/>**User impact:** For users with ETCD as the KVDB and aggressive ETCD compactions, some of the nodes in Portworx cluster skipped the automatic application of updated licenses. Users had to restart the Portworx service to pick up the changes.<br/><br/>**Resolution:** Portworx no longer skips license updates when aggressive ETCD compactions are used.| 

> | PWX-16793 | This fix speeds up Portworx installations for hosts that did not have NFS services preinstalled. <br/><br/>**User impact:** Some host environments may have a large amount of package repositories or have a slow internet connection, which could lead into timeouts while installing NFS services during Portworx installation or upgrade, which ultimately results in inability to use SharedV4 volumes that depend on NFS. <br/><br/>**Resolution:** Portworx installs have been sped up and avoid timeouts with NFS services installations. |

Reviewer 5: ```suggestion
| PWX-16793 | Sometimes Portworx installation timed out while installing NFS packages. <br/><br/>**User impact:** Some host environments may have a large amount of package repositories or have a slow internet connection, which could lead into timeouts while installing NFS services during Portworx installation or upgrade, which ultimately results in inability to use SharedV4 volumes that depend on NFS. <br/><br/>**Resolution:** Portworx installs have been sped up and avoid timeouts with NFS services installations. |
```

| PWX-16775 | The Google object store did not have pagination for object enumeration, which caused any list call to list everything in the bucket.<br/><br/>**User impact:** Cloudsnap backups and restores failed to start and the request timed out. Listing cloudsnaps through `pxctl` also timed-out.<br/><br/>**Resolution:** Added pagination to object enumeration with the Google object store. |
| PWX-16796 | Proxy username and password were ignored as part of `PX_HTTP_PROXY` on Portworx Essentials causing license renewal to fail. <br/><br/>**User impact:** Portworx essentials clusters went into "license expired" mode with PX_HTTP_PROXY <br/><br/>**Resolution:** Portworx essentials now honors Username and Password fields given as part of `PX_HTTP_PROXY` to successfully make connections with proxy |

> | PWX-16072 | When Portworx was installed in PAYG mode, the cluster license  expired after being unable to reach to billing server instead of going into maintenance mode. <br/><br/>**User Impact:** When a PAYG node license expired, users had to recommission the node. <br/><br/>**Resolution:** Portworx PAYG nodes now enter maintenance mode correctly when the billing server is unreachable.| 

Reviewer 5: ```suggestion
| PWX-16072 | When Portworx was installed in PAYG mode, the cluster license expired after being unable to connect to that billing server instead of going into maintenance mode. <br/><br/>**User Impact:** When a PAYG node license expired, users had to recommission the node. <br/><br/>**Resolution:** Portworx PAYG nodes now enter maintenance mode correctly when the billing server is unreachable.| 
```

> | PWX-16429 | Adding a new drive using `pxctl service drive add` was failing due to an issue with applying labels on the new pool. <br/><br/>**User impact:** If users wanted to add a new Portworx pool to the node, their command to add a new drive using `pxctl service drive add` failed with an error message about labels being too long. This prevented users from creating new pools. <br/><br/>**Resolution:** The Kubernetes labels are being skipped to be applied to the backend storage pools. |

Reviewer 5: ```suggestion
| PWX-16429 | Adding a new drive using the `pxctl service drive add` command was failing due to an issue with applying labels on the new pool. <br/><br/>**User impact:** If users wanted to add a new Portworx pool to the node, their command to add a new drive using `pxctl service drive add` failed with an error message about labels being too long. This prevented users from creating new pools. <br/><br/>**Resolution:** The Kubernetes labels are being skipped to be applied to the backend storage pools. |
```

> | PWX-16206 | Portworx failed to correctly detect the max storage nodes per zone when running on GKE. <br/><br/>**User impact:** When running on GKE with autoscaling enabled, Portworx did not detect the preferred value to use for max storage nodes per zone for it's cloud drives. As a result, Portworx would run without or with an incorrect max storage nodes per zone value. This resulted in issues when the cluster size was scaled and unintended nodes became become storage nodes. <br/><br/>**Resolution:** The calculation for max storage nodes per zone based on the GKE autoscaling min pool size values has been fixed. In addition to this, if the minimum number of nodes in the cluster is lower than total number of zones, at least one Portworx node will now be made storage node in a given zone. |

Reviewer 5: ```suggestion
| PWX-16206 | Portworx failed to correctly detect the value of the `maxStorageNodesPerZone` parameter when running on GKE. <br/><br/>**User impact:** When running on GKE with autoscaling enabled, Portworx did not detect the preferred value to use for the `maxStorageNodesPerZone` parameter for its cloud drives. As a result, Portworx would run without or with an incorrect value for the `maxStorageNodesPerZone` parameter. This resulted in issues when the cluster size was scaled and unintended nodes became become storage nodes. <br/><br/>**Resolution:** The calculation for the value of the `maxStorageNodesPerZone`parameter on the GKE autoscaling min pool size values has been fixed. In addition to this, if the minimum number of nodes in the cluster is lower than the total number of zones, and at least one Portworx node will now be made a storage node in a given zone. |
```

> | PWX-16554 | An incorrect check prevented HA level restoration for volumes with HA level 3 under some conditions during decommission operation.<br/<br/> **User impact:** Decommission operation failed to restore HA level for affected volumes if the volume had HA level 3 under some conditions. For such volumes, restore HA level using the pxctl CLI command for ha-update. | 

Reviewer 5: ```suggestion
| PWX-16554 | An incorrect check prevented HA level restoration for volumes with HA level 3 under some conditions during decommission operation.<br/<br/> **User impact:** Decommission operation failed to restore HA level for affected volumes if the volume had HA level 3 under some conditions. </br></br> ** Resolution:** When you decommission a node, Portworx now properly restores the HA level.  | 
```

| PWX-16407 | DR migration sometimes became stuck in the "Active" state when Portworx restarted while migration was beginning . <br/><br/>**User impact:** Any additional DRs also became stuck and did not finish. <br/><br/>**Resolution:** Portworx now handles this situation better.| 
| PWX-16495 | Currently, the KVDB lock is held over the Inspect API. If the remote node didn't respond, Portworx held the KVDB lock for more than 3 minutes and the node where the Attach was issued asserted. <br/><br/> **User impact:** The Portworx service could assert and restart if it tried to remotely detach a volume from a peer node but the request to do that took more than 3 minutes. <br/><br/>**Resolution:** Portworx will not assert and restart if such a request gets stuck or does not return within a given timeout. |

> | PWX-13527 | Internal kvdb startup would fail with the error : "Operation cannot be fulfilled on configmaps". This would usually require a wipe of the node and re-install. <br/><br/>**Resolution**: Portworx will now detect such errors received from Kubnernetes and will retry the operation instead of exiting.	| 

Reviewer 5: ```suggestion
| PWX-13527 | Internal KVDB startup would fail with the error : "Operation cannot be fulfilled on configmaps". This would usually require a wipe of the node and re-install. <br/><br/>**Resolution**: Portworx will now detect such errors received from Kubnernetes and will retry the operation instead of exiting.	| 
```

| PWX-16384 | When a node with a sharedv4 encrypted volume attached was rebooted, the volume was not re-exported over NFS for other nodes to consume. <br/><br/>**User impact:** Since it's an encrypted volume, it couldn't be attached without a passphrase. <br/><br/>**Resolution:** Portworx now triggers a restart of the app, which will reattach the encrypted sharedv4 volume. |

> | PWX-16729 | A particular race condition caused an Unmount of a sharedv4 volume to succeed without actually removing the underlying NFS mountpoint. <br/><br/>**User impact:** This caused the pod using the sharedv4 volume to be stuck in the "Terminating" state. <br/><br/>**Resolution:** Portworx no longer experiences this race condition. |

Reviewer 5: ```suggestion
| PWX-16729 | A particular race condition caused an unmount of a sharedv4 volume to succeed without actually removing the underlying NFS mountpoint. <br/><br/>**User impact:** This caused the pod using the sharedv4 volume to be stuck in the "Terminating" state. <br/><br/>**Resolution:** Portworx no longer experiences this race condition. |
```

| PWX-16715  |In certain cloud deployments, API calls to the instance metadata service or the cloud management portals are blocked or routed through a proxy. <br/><br/>**User impact:** In these cases, Portworx calls to the cloud were blocked indefinitely, causing Portworx to fail to initialize. <br/><br/>**Resolution:** Portworx now invokes all the cloud and instance metadata APIs with a timeout and will avoid getting blocked indefinitely. | 
| PWX-14101  | For certain providers like vSphere, when a cloud drive created by Portworx was deleted out of band, Portworx ignored it, created a new disk, and started as a brand new node. <br/><br/>**User impact:** This caused an issue if there were no additional licenses in the cluster. <br/><br/>**Resolution:** If a disk is deleted out of band or is moved to another datastore in vShpere, Portworx now errors-out and does not create new drives.| 

> | PWX-16465 | NEED DESCRIPTION <br/><br/>**User impact:** Encrypted volume attach operations sometimes failed and caused Portworx to assert. The attach call risked getting stuck in the "dm-crypt" layer or in an HTTPS API to the KMS provider.<br/><br/>**Resolution:** Portworx will no longer assert if any calls get stuck in the "dm-crypt" layer or if the HTTPS calls to the KMS providers timeout. |

Reviewer 6: Description: A lock was being held while performing operations in dm-crypt layer or while making an external KMS API for encrypted volumes. If either of them would take longer than expected time, PX would assert.
User Impact: PX would restart in certain scenarios while attaching encrypted volumes.

> | PWX-15043  | For NFS Proxy Volumes in the `pxctl volume inspect` command output display the "Mount Options" field. <br/><br/>**User Impact:** the `pxctl volume inspect` command output didn't show the "Mount Options" field for NFS proxy volumes, even if the mount options were explicitly provided while creating such volume. | 

Reviewer 5: ```suggestion
| PWX-15043  |  The `pxctl volume inspect` command output did not show the "Mount Options" field for NFS proxy volumes. <br/><br/>**User Impact:** You could not see the "Mount Options" field for NFS proxy volumes, even if you explicitly provided the mount options while creating such a volume. <br/><br/>**Resolution:** The `pxctl volume inspect` command output now shows the "Mount Options" field for NFS proxy volumes. | 
```
